File: app/src/main/java/com/fatbook/fatbookapp/ui/listeners/OnRecipeClickListener.java
package com.fatbook.fatbookapp.ui.listeners;

import com.fatbook.fatbookapp.core.Recipe;

public interface OnRecipeClickListener {
    void onRecipeClick(int position);
    void onBookmarksClick(Recipe recipe, boolean add);
    void onForkClicked(Recipe recipe, boolean fork, int position);
}

File: app/src/main/java/online/fatbook/fatbookapp/ui/base/OnRecipeClickListener.kt
package online.fatbook.fatbookapp.ui.base

import online.fatbook.fatbookapp.core.recipe.RecipeSimpleObject
import online.fatbook.fatbookapp.ui.feed.adapters.RecipeAdapter

interface OnRecipeClickListener {
    fun onRecipeClick(id: Long)
    fun onBookmarksClick(recipe: RecipeSimpleObject?, bookmark: Boolean, position: Int)
    fun onForkClicked(recipe: RecipeSimpleObject?, fork: Boolean, position: Int)
    fun onForkClicked(
        recipe: RecipeSimpleObject?,
        fork: Boolean,
        position: Int,
        recipePreviewItemViewHolder: RecipeAdapter.RecipePreviewItemViewHolder
    )
    fun onUsernameClick(username: String)

}